pages/form/returning-members.js

```diff
@@ -96,96 +93,86 @@ const CONTINUING_STATUS = [
-    termDescription: false,
-    previousTerms: false,
-    futureTerms: false,
-    activeStatus: false,
+    nextSchoolTerm: false,
+    nextTermActivity: false,
     subteam: false,
-    activeStatus: false,
-    continuingStatus: false,
+    nextTermRole: false,
     email: false,
   });
 
-  const handleSave = () => {
-    const updatedErrorList = { ...hasError };
+  const setErrorMessages = (formErrors) => {
+    validateField(formValues, formErrors, 'fullName');
+    validateField(formValues, formErrors, 'email');
+    validateField(formValues, formErrors, 'nextSchoolTerm');
+    validateField(formValues, formErrors, 'subteam');
+    validateField(formValues, formErrors, 'nextTermRole');
+    validateField(formValues, formErrors, 'nextTermActivity');
 
-    // this resets all fields to display no errors initially, the errors will update when correcting some fields
-    for (const [name, value] of Object.entries(updatedErrorList)) {
-      updatedErrorList[name] = false;
-    }
+    setHasError(formErrors);
+  }
 
-    if (!formValues.fullName || formValues.fullName.split(' ').length !== 2) {
-      updatedErrorList['fullName'] = true;
-    }
-    if (!formValues.email || !isEmail(formValues.email)) {
-      updatedErrorList['email'] = true;
-    }
-    if (!formValues.termDescription) {
-      updatedErrorList['termDescription'] = true;
-    }
-    if (!formValues.subteam) {
-      updatedErrorList['subteam'] = true;
-    }
-    if (!formValues.continuingStatus) {
-      updatedErrorList['continuingStatus'] = true;
-    }
-    if (!formValues.activeStatus) {
-      updatedErrorList['activeStatus'] = true;
-    }
-    if (!formValues.futureTerms) {
-      updatedErrorList['futureTerms'] = true;
-    }
+  const handleSubmit = (evt) => {
+    evt.preventDefault();
+    const formErrors = { ...hasError };
+
+    clearErrorMessages(formErrors);
+    setErrorMessages(formErrors);
+
+    const formHasErrors = Object.values(formErrors).some(err => err);
+    if (!formHasErrors) {
+      const {fullName, nextSchoolTerm, previousTerms, futureTerms, subteam, nextTermActivity,
+              nextTermRole, email, termComments, desiredWork} = formValues;
 
-    setHasError(updatedErrorList);
-    const submissionHasErrors = Object.values(updatedErrorList).some(
-      (hasError) => hasError === true
-    );
-    if (submissionHasErrors) {
-      return;
+      const fullNameParts = fullName.split(/\s+/);
+      updateUser(dispatch, {
+          name: {
+              first: fullNameParts[0].trim(),
+              last: fullNameParts[fullNameParts.length - 1].trim()
+          },
+          personalEmail: email.trim(),
+          subteams: [subteam], // NOTE: As of March 2022, members can only select one option for subteam. Before this, members can select multiple subteams. We will keep subteams as an array for now for backwards-compatability and to prevent conflicts with Database data.
+          activeSchoolTerms: [...previousTerms, ...futureTerms],
```

Comment:
Like I mentioned previously, I don't know if we should be modifying this through the user's form input. I think that when the user submits the form, the date when they submitted should map to the appropriate term then this term should get **appended** to `activeSchoolTerms`.